pkg/kubernetes/sort.go

```diff
@@ -0,0 +1,130 @@
+		strings.ToLower("MutatingWebhookConfiguration"):   admissionPriority,
+		strings.ToLower("Namespace"):                      namespacePriority,
+		strings.ToLower("NetworkPolicy"):                  networkResourcePriority,
+		strings.ToLower("PersistentVolumeClaim"):          mountableDataPriority,
+		strings.ToLower("PersistentVolume"):               mountableDataBackingResourcePriority,
+		strings.ToLower("PodDisruptionBudget"):            pdbPriority,
+		strings.ToLower("Pod"):                            workloadPriority,
+		strings.ToLower("PodPreset"):                      workloadModifierPriority,
+		strings.ToLower("PodSecurityPolicy"):              workloadModifierPriority,
+		strings.ToLower("ReplicaSet"):                     workloadControllerPriority,
+		strings.ToLower("RoleBinding"):                    roleBindingPriority,
+		strings.ToLower("Role"):                           rolePriority,
+		strings.ToLower("Secret"):                         mountableDataPriority,
+		strings.ToLower("ServiceAccount"):                 serviceAccountPriority,
+		strings.ToLower("Service"):                        networkResourcePriority,
+		strings.ToLower("StatefulSet"):                    workloadControllerPriority,
+		strings.ToLower("StorageClass"):                   storageClassPriority,
+		strings.ToLower("UnregisteredClusterResource"):    lowestPriority,
+		strings.ToLower("ValidatingWebhookConfiguration"): admissionPriority,
+	}
+)
+
+// SortManifests takes in a list of manifests and sorts them by the priority of their kind.
+// The kind's priorities are defined above in the var 'priorities'. Lower numbered priorities
+// should be deployed first.
+func (c *controller) SortManifests(manifests []map[string]interface{}) ([]map[string]interface{}, error) {
+	// Map of priorities to lists of manifests.
+	manifestMap := map[int][]map[string]interface{}{
+		0:    []map[string]interface{}{},
+		20:   []map[string]interface{}{},
+		30:   []map[string]interface{}{},
+		40:   []map[string]interface{}{},
+		50:   []map[string]interface{}{},
+		70:   []map[string]interface{}{},
+		80:   []map[string]interface{}{},
+		90:   []map[string]interface{}{},
+		100:  []map[string]interface{}{},
+		110:  []map[string]interface{}{},
+		1000: []map[string]interface{}{},
+	}
+
+	for _, manifest := range manifests {
+		u, err := c.ToUnstructured(manifest)
+		if err != nil {
+			return nil, fmt.Errorf("kubernetes: error sorting manifests: %w", err)
+		}
+
+		if _, ok := priorities[strings.ToLower(u.GetKind())]; ok {
+			priority := priorities[strings.ToLower(u.GetKind())]
+			s := manifestMap[priority]
+			s = append(s, manifest)
+			manifestMap[priority] = s
+		} else {
+			s := manifestMap[lowestPriority]
+			s = append(s, manifest)
+			manifestMap[lowestPriority] = s
+		}
+	}
+
+	// Store the keys in slice in sorted asc order.
+	keys := make([]int, len(manifestMap))
+	i := 0
+
+	for k := range manifestMap {
+		keys[i] = k
+		i++
+	}
+
+	sort.Ints(keys)
```

Comment:
Do you need to do that, since when you created that it was already sorted then you appended to the values?

Can you try to comment it out and run the tests?

Reply:
Actually yes, because [Go maps are unordered](https://www.geeksforgeeks.org/golang-maps/).

Reply:
In other words, we can't rely on the maps to remain some order providing their keys.

Reply:
For more specifics, see https://stackoverflow.com/questions/18342784/how-to-iterate-through-a-map-in-golang-in-order